app/renderer/reducers/frameReducer.js

```diff
@@ -112,26 +115,28 @@ const frameReducer = (state, action, immutableAction) => {
       }
 
       // TODO fix race condition in Muon more info in #9000
-      const title = immutableAction.getIn(['tabValue', 'title'])
+      const title = tab.get('title')
       if (title != null) {
         state = state.setIn(['frames', index, 'title'], title)
       }
 
       // TODO fix race condition in Muon more info in #9000
```

Comment:
yeah I think that race condition is no longer present, that's why I suggested to do a follow up, remove todos and close issue #9000